ArduPlane/quadplane.cpp

```diff
@@ -4309,4 +4316,45 @@ bool QuadPlane::landing_with_fixed_wing_spiral_approach(void) const
              cmd.p1 == NAV_VTOL_LAND_OPTIONS_FW_SPIRAL_APPROACH));
 }
 
+/*
+  setup scaling of roll and pitch angle P gains to match fixed wing gains
+
+  we setup the angle P gain to match fixed wing at high speed (above
+  ARSPD_FBW_MIN) where fixed wing surfaces are presumed to
+  dominate. At lower speeds we use the multicopter angle P gains.
+*/
+void QuadPlane::setup_rp_fw_angle_gains(void)
+{
+    const float mc_angR = attitude_control->get_angle_roll_p().kP();
+    const float mc_angP = attitude_control->get_angle_pitch_p().kP();
+    const float fw_angR = 1.0/plane.rollController.tau();
+    const float fw_angP = 1.0/plane.pitchController.tau();
+
+    if (!is_positive(mc_angR) || !is_positive(mc_angP)) {
+        // bad configuration, don't scale
+        return;
+    }
+
+    float aspeed;
+    if (!ahrs.airspeed_estimate(aspeed)) {
+        // can't get airspeed, no scaling of VTOL angle gains
+        return;
+    }
+
+    const float low_airspeed = 3.0;
```

Comment:
I wonder if we should use the VTOL way-point speed here. The presumption being that the aircraft will be stable in 100% VTOL at that speed. Would be nice to remove that jump up to 1:1 scaling when we move out of POS1.

However if we start using this more universally including VTOL modes this might allow pushing WP speed higher and then we have a odd circular dependency. I guess we could always add a new param in the future.

Reply:
users don't tend to test if their vehicle is stable at the max WP speed. I think using 3 works well

Reply:
I don't care much but I agree with Pete here. You could set the WP speed default to 3.